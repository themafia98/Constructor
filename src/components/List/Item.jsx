import React, {useState} from 'react';

function Item(props) {
    const [id] = useState(props.id);
    const [item] = useState({...props.item});
    return (
        <div className = 'item' data-id = {id}>{item.name}</div>
    )
}

export default Item;