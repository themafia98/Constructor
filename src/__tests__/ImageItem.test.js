import React from 'react';
import renderer from 'react-test-renderer';

import ImageItem from '../components/imageViewer/imageItem';
import Item from '../components/List/Item';

const path = "https://images.unsplash.com/";
let imageurl = "photo-1504215680853-026ed2a45def?ixlib=rb-1.2.1&amp;q=80&amp;fm=jpg&amp;";
imageurl = imageurl + "crop=entropy&amp;cs=tinysrgb&amp;w=1080&amp;fit=max&amp;ixid=eyJhcHBfaWQiOjgwNjg0fQ";

const imagePath = path + imageurl;

const ImageItemComponent = renderer.create(
    <ImageItem id = {0} urls ={{regular: imagePath, urlFull: imagePath}} />
)

test("ImageItem.js test", function(){

    let snapshotImageItem = ImageItemComponent.toJSON();
    expect(snapshotImageItem).toMatchSnapshot();

    const eventClick = ImageItemComponent.root.find(item => item.props.className === 'ImageItem');
    eventClick.props.onClick({stopPropagation: () => null});

    snapshotImageItem = ImageItemComponent.toJSON();
    expect(snapshotImageItem).toMatchSnapshot();

});