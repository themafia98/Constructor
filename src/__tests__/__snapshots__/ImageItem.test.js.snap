// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageItem.js test 1`] = `
<div
  className="ItemBox"
>
  <img
    alt="item"
    className="ImageItem"
    onClick={[Function]}
    src="https://images.unsplash.com/photo-1504215680853-026ed2a45def?ixlib=rb-1.2.1&amp;q=80&amp;fm=jpg&amp;crop=entropy&amp;cs=tinysrgb&amp;w=1080&amp;fit=max&amp;ixid=eyJhcHBfaWQiOjgwNjg0fQ"
  />
</div>
`;

exports[`ImageItem.js test 2`] = `
<div
  className="ItemBox"
>
  <img
    alt="item"
    className="ImageItem"
    onClick={[Function]}
    src="https://images.unsplash.com/photo-1504215680853-026ed2a45def?ixlib=rb-1.2.1&amp;q=80&amp;fm=jpg&amp;crop=entropy&amp;cs=tinysrgb&amp;w=1080&amp;fit=max&amp;ixid=eyJhcHBfaWQiOjgwNjg0fQ"
  />
</div>
`;
